packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -1,211 +1,359 @@
+    },
+    [apiRef, logger, options.editMode],
+  );
+
   const onEditRowModelChange = React.useCallback(
     (handler: (param: GridEditRowModelParams) => void): (() => void) => {
-      return apiRef.current.subscribeEvent(GRID_EDIT_ROW_MODEL_CHANGE, handler);
+      return apiRef.current.subscribeEvent(GRID_ROW_EDIT_MODEL_CHANGE, handler);
     },
     [apiRef],
   );
+
   const onCellModeChange = React.useCallback(
     (handler: (param: GridCellModeChangeParams) => void): (() => void) => {
       return apiRef.current.subscribeEvent(GRID_CELL_MODE_CHANGE, handler);
     },
     [apiRef],
   );
+
   const onEditCellChange = React.useCallback(
-    (handler: (param: GridEditCellParams) => void): (() => void) => {
-      return apiRef.current.subscribeEvent(GRID_CELL_CHANGE, handler);
+    (handler: (param: GridEditCellValueParams) => void): (() => void) => {
+      return apiRef.current.subscribeEvent(GRID_CELL_EDIT_PROPS_CHANGE, handler);
     },
     [apiRef],
   );
+
   const onEditCellChangeCommitted = React.useCallback(
-    (handler: (param: GridEditCellParams) => void): (() => void) => {
-      return apiRef.current.subscribeEvent(GRID_CELL_CHANGE_COMMITTED, handler);
+    (handler: (param: GridEditCellValueParams) => void): (() => void) => {
+      return apiRef.current.subscribeEvent(GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED, handler);
     },
     [apiRef],
   );
 
-  useGridApiEventHandler(apiRef, GRID_CELL_CHANGE, options.onEditCellChange);
-  useGridApiEventHandler(apiRef, GRID_CELL_CHANGE_COMMITTED, options.onEditCellChangeCommitted);
+  const handleExitEdit = React.useCallback(
+    (params: GridCellParams, event: React.SyntheticEvent) => {
+      setCellMode(params.id, params.field, 'view');
+
+      if (!isKeyboardEvent(event)) {
+        return;
+      }
+
+      if (isCellEditCommitKeys(event.key)) {
+        apiRef.current.publishEvent(GRID_CELL_NAVIGATION_KEYDOWN, params, event);
+        return;
+      }
+      if (isEscapeKey(event.key) || isDeleteKeys(event.key)) {
+        apiRef.current.setCellFocus(params);
+      }
+    },
+    [apiRef, setCellMode],
+  );
+
+  const handleEnterEdit = React.useCallback(
+    (params: GridCellParams, event: React.MouseEvent | React.KeyboardEvent) => {
+      if (!params.isEditable) {
+        return;
+      }
+      if (isKeyboardEvent(event) && isAlphaKeys(event.key)) {
+        const propsParams = apiRef.current.getEditCellPropsParams(params.id, params.field);
+        propsParams.props.value = '';
+        apiRef.current.setEditCellProps(propsParams);
+      }
+      setCellMode(params.id, params.field, 'edit');
+    },
+    [apiRef, setCellMode],
+  );
+
+  const preventTextSelection = React.useCallback((params, event) => {
+    const isMoreThanOneClick = event.detail > 1;
+    if (params.isEditable && isMoreThanOneClick) {
+      // If we click more than one time, then we prevent the default behavior of selecting the text cell.
+      event.preventDefault();
```

Comment:
This breaks the double click to select inside an `<input>`. A end-users should be able to select all the text with a double click.

Reply:
Ah yes! Nicely spotted